keyboards/ploopyco/ploopyco.c

```diff
@@ -127,30 +130,26 @@ void encoder_driver_task(void) {
     lastScroll = timer_read();
 }
 #endif
-
 report_mouse_t pointing_device_task_kb(report_mouse_t mouse_report) {
     if (is_drag_scroll) {
-        scroll_accumulated_h += (float)mouse_report.x / PLOOPY_DRAGSCROLL_DIVISOR_H;
-        scroll_accumulated_v += (float)mouse_report.y / PLOOPY_DRAGSCROLL_DIVISOR_V;
-
-        // Assign integer parts of accumulated scroll values to the mouse report
-        mouse_report.h = (int8_t)scroll_accumulated_h;
+        // Assign integer parts of accumulated scroll values with remainders to the mouse report
+        mouse_report.h = (mouse_report.x + scroll_remainder_h) / PLOOPY_DRAGSCROLL_DIVISOR_H;
 #ifdef PLOOPY_DRAGSCROLL_INVERT
-        mouse_report.v = -(int8_t)scroll_accumulated_v;
+        mouse_report.v = -(mouse_report.y + scroll_remainder_v)  / PLOOPY_DRAGSCROLL_DIVISOR_V;
 #else
-        mouse_report.v = (int8_t)scroll_accumulated_v;
+        mouse_report.v = (mouse_report.y + scroll_remainder_v)  / PLOOPY_DRAGSCROLL_DIVISOR_V;
 #endif
 
-        // Update accumulated scroll values by subtracting the integer parts
-        scroll_accumulated_h -= (int8_t)scroll_accumulated_h;
-        scroll_accumulated_v -= (int8_t)scroll_accumulated_v;
+        // Update accumulated scroll values to the remainders of mouse report
+        scroll_remainder_h = (mouse_report.x + scroll_remainder_h) % PLOOPY_DRAGSCROLL_DIVISOR_H;
+        scroll_remainder_v = (mouse_report.y + scroll_remainder_v) % PLOOPY_DRAGSCROLL_DIVISOR_V;
 
         // Clear the X and Y values of the mouse report
         mouse_report.x = 0;
         mouse_report.y = 0;
-
-        mouse_report.x = 0;
-        mouse_report.y = 0;
+    } else {
+        scroll_remainder_h = 0;
+        scroll_remainder_v = 0;
```

Comment:
This is so the remainders from a previous drag scroll use aren't at the start of a new one.